### PR TITLE
Annual Receipts UI now connects to addon

### LINKED ISSUE #5603: Annual Receipts tab UI should connect to addon
## User Story

As a user, I want to view my annual receipts from the Donor Profile app so that I can find all my annual receipts in one familiar place.

## Details

The endpoint needed to fetch dynamic annual receipt data should be registered via a new Tab in the Annual Receipts addon. For that reason, this PR is dependent upon work in the Annual Receipts addon.

## Additional Context

Figma designs: https://www.figma.com/file/BKr2K0XdDyRRoccJbJKRD7/Epic%3A-Donor-Profile?node-id=397%3A1743

## Acceptance Criteria

-   [x] Annual Receipts can be viewed in the Donor Profile app
-   [x] Annual Receipts tab content is dynamically sourced

### PR DESCRIPTION
**Pull request by henryholtgeerts**
**_NOTE: This PR is dependent up work done in this Give Annual Receipts PR: https://github.com/impress-org/give-annual-receipts/pull/89_**

Resolves #5603

## Description

This PR introduces improvements to the Annual Receipts tab UI, so that it can dynamically populate the Annual Receipts table using data fetched from the Annual Receipts addon. 

## Affects

This PR primarily affects frontend rendering of the Donor Profiles application, and specifically the Annual Receipts tab.

## Visuals

Screen Shot 2021-02-17 at 4 42 40 PM

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

### CODE REVIEW

File: src/DonorProfiles/resources/js/app/tabs/annual-receipts/content.js [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+			<Heading>
+				{ __( 'Loading...', 'give' ) }
+			</Heading>
+			<AnnualReceiptTable />
+		</Fragment>
+	) : (
 		<Fragment>
 			<Heading>
-				Annual Receipts
+				{ annualReceipts ? `${ Object.entries( annualReceipts ).length } ${ __( 'Total Annual Receipts', 'give' ) }` : __( '0 Total Annual Receipts', 'give' ) }
```
Comment (kjohnson) [inline src/DonorProfiles/resources/js/app/tabs/annual-receipts/content.js:29]: I would suggest using a single translatable string and use `sprintf` to insert the number.

**Review by kjohnson — CHANGES_REQUESTED** [review]
Works as expected.

However, the tab should only show if the add-on is installed, active, and the feature enabled.

File: src/DonorProfiles/resources/js/app/tabs/annual-receipts/content.js (hunk 1 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/tabs/annual-receipts/content.js]: Hmm -- without adding another JS library to support a formatter like sprintf, I've opted to abstract the annual receipts count determination logic into it's own line during the render method. This ensures that only 1 translation is needed. Ref: https://github.com/impress-org/givewp/pull/5611/commits/6b4315f1bf401bf8712084a2cce674bd44ff3e82

**Comment by henryholtgeerts** [comment]
@kjohnson Thanks for the review! Now that this branch has been updated from the Epic branch, logic to ensure that tabs only display when the addon is enabled has been introduced and works as expected. I've also addressed your concern about duplicate translations.

**Review by kjohnson — APPROVED** [review]
Looks good. We probably need to consider how we handle singular/plural translations in WordPress JavaScript. On the server we have `_n()`, but I'm not seeing a JS version - need to look into that more, but that can be a separate change.

**Pull request merged**